src/Core/Grid/Definition/Factory/AuthorizedApplicationGridDefinitionFactory.php

```diff
@@ -0,0 +1,132 @@
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Definition\Factory;
+
+use PrestaShop\PrestaShop\Core\Grid\Action\Row\RowActionCollection;
+use PrestaShop\PrestaShop\Core\Grid\Action\Row\Type\LinkRowAction;
+use PrestaShop\PrestaShop\Core\Grid\Column\ColumnCollection;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\ActionColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\CustomTypeColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\LinkColumn;
+
+/**
+ * Class AuthorizedApplicationGridDefinitionFactory is responsible for creating new instance of Authorized Application grid definition.
+ */
+final class AuthorizedApplicationGridDefinitionFactory extends AbstractGridDefinitionFactory
+{
+    use DeleteActionTrait;
+
+    public const GRID_ID = 'authorized_application';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getId()
+    {
+        return self::GRID_ID;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getName()
+    {
+        return $this->trans('List of applications', [], 'Admin.Navigation.Menu');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getColumns()
+    {
+        $columns = (new ColumnCollection())
+            ->add(
+                (new LinkColumn('name'))
+                    ->setName($this->trans('Name', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'name',
+                        'route' => 'admin_authorized_applications_view',
+                        'route_param_name' => 'authorizedApplicationId',
+                        'route_param_field' => 'id_authorized_application',
+                    ])
+            )
+            ->add(
+                (new CustomTypeColumn('api_accesses_states'))
+                    ->setName($this->trans('API access state', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'apiAccesses',
```

Comment:
Now that's me that have a problem with this `CustomTypeColumn` 😅
I feel like we're losing the option type verifications. For now, for this particular use case it's okay because we only need the `field` option. But what about a use case where we need another option, like a list for a select element? We can add the option to the column type but then we cannot use it without the new option.

Reply:
Ok after discussion with @jolelievre and @atomiix we decide to use a specific column but in the right namespace. cc @FabienPapet

Reply:
I still don't think this is the right namespace 🤣.  But this is subjective and non-blocking. (Soo many worse things than a misplaced class)